src/Validation/Validator/StringValidator.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ScrumWorks\OpenApiSchema\Validation\Validator;
+
+use RuntimeException;
+use ScrumWorks\OpenApiSchema\Validation\BreadCrumbPath;
+use ScrumWorks\OpenApiSchema\Validation\Result\ValidationResultBuilderFactoryInterface;
+use ScrumWorks\OpenApiSchema\Validation\Result\ValidationResultBuilderInterface;
+use ScrumWorks\OpenApiSchema\ValueSchema\StringSchema;
+
+final class StringValidator extends AbstractValidator
+{
+    private StringSchema $schema;
+
+    public function __construct(
+        ValidationResultBuilderFactoryInterface $validationResultBuilderFactory,
+        StringSchema $schema
+    ) {
+        parent::__construct($validationResultBuilderFactory, $schema);
+
+        $this->schema = $schema;
+    }
+
+    protected function doValidation(
+        ValidationResultBuilderInterface $resultBuilder,
+        $data,
+        BreadCrumbPath $breadCrumbPath
+    ): void {
+        if (! $this->validateNullable($resultBuilder, $data, $breadCrumbPath)) {
+            return;
+        }
+
+        if (! \is_string($data)) {
+            $resultBuilder->addTypeViolation('string', $breadCrumbPath);
+            return;
+        }
+
+        $strlen = $this->schema->getFormat() === 'binary' ? \strlen($data) : \mb_strlen($data);
+
+        if (($minLength = $this->schema->getMinLength()) !== null && $minLength >= $strlen) {
+            $resultBuilder->addMinLengthViolation($minLength, $breadCrumbPath);
+        }
+        if (($maxLength = $this->schema->getMaxLength()) !== null && $maxLength <= $strlen) {
+            $resultBuilder->addMaxLengthViolation($maxLength, $breadCrumbPath);
+        }
+        if (($pattern = $this->schema->getPattern()) !== null && ! \preg_match("~${pattern}~", $data)) {
```

Comment:
tak ale tady uz se tomu nevyhnes! :D

`\preg_match('~' . \ str_replace('~', '\\~', $pattern) . '~', $data)` - teda alespon doufam ze jsem to napsal dobre, nezkousel jsem to...

navic teda podle dokumentace by se pro patterny meli pouzit regexpy z ECMA 262, tj https://www.ecma-international.org/ecma-262/5.1/#sec-15.10.1 - tim jne narazim na to ze PCRE regexpy nabizeji mnohem mnohem sirsi moznosti

Reply:
navic teda to volani je tady dost na hrane, nebot vubec nechytne Errors z preg_match napr. ohledne spatne syntaxe regexpu -viz vsechny ty srandy co se resi v Nette Utils String u regularu... - mimojine to by pripadne byla moznost, protoze proeprty-reader uz ma stejne nette utils jako zavislost, tak ji dat i sem a vyuzit primo nette\utils\stings